src/app/shared/team-radio-switch/team-radio-switch.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, Input, forwardRef, ElementRef } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Team } from './team-radio-switch.model';
+
+@Component({
+  selector: 'app-team-radio-switch',
+  templateUrl: './team-radio-switch.component.html',
+  styleUrls: ['./team-radio-switch.component.css'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => TeamRadioSwitchComponent),
+      multi: true,
+    },
+  ],
+})
+export class TeamRadioSwitchComponent implements ControlValueAccessor {
+  @Input() team1!: Team;
+  @Input() team2!: Team;
+
+  onChange: any = () => {};
+  onTouched: any = () => {};
+
+  private teamId: number = 0;
+
+  public get valueId(): number {
```

Comment:
We've already discussed the necessity of this access modifiers

Reply:
done